client/my-sites/pages/page.jsx

```diff
@@ -134,6 +138,10 @@ const Page = React.createClass( {
 			return null;
 		}
 
+		if ( this.props.hasStaticFrontPage && this.props.isPostsPage ) {
```

Comment:
> Do we need to check both of these?

Unfortunately, it looks like we do. This is a "valid" setting in wp-admin:
<img width="528" alt="screen shot 2016-10-20 at 9 20 26 am" src="https://cloud.githubusercontent.com/assets/1587282/19561394/8a57f594-96a6-11e6-80dc-e5d5eff70c8e.png">

That results in the front page showing posts as well as the dummy page.